packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx

```diff
@@ -465,3 +432,102 @@ const getError = (
     "Unknown error"
   );
 };
+
+function useInitialFormValues({
+  eventType,
+  rescheduleUid,
+  isRescheduling,
+}: {
+  eventType: ReturnType<typeof useEvent>["data"];
+  rescheduleUid: string | null;
+  isRescheduling: boolean;
+}) {
+  const [initialValues, setDefaultValues] = useState<DefaultValues>({});
+  const bookingData = useBookerStore((state) => state.bookingData);
+  const formValues = useBookerStore((state) => state.formValues);
+  const searchParams = useSearchParams();
+  const routerQuery = useRouterQuery();
+  const session = useSession();
+  useEffect(() => {
+    (async function () {
+      if (Object.keys(formValues).length) return formValues;
+
+      if (!eventType?.bookingFields) {
+        return {};
+      }
+      const querySchema = getBookingResponsesPartialSchema({
+        eventType: {
+          bookingFields: eventType.bookingFields,
+        },
+        view: rescheduleUid ? "reschedule" : "booking",
+      });
+
+      const parsedQuery = await querySchema.parseAsync({
```

Comment:
`parseAsync` is required because `getBookingResponsesPartialSchema` gives async schema